lib/mount.js

```diff
@@ -22,11 +22,21 @@ const mount = (app, { errLogger, logger }={}) =>
       .then(log(logger, { req, res }))
 
 const write = (req, res) => ({ body, headers={}, statusCode=200 }) => {
-  res.statusCode = statusCode
-
   for (var name in headers)
     res.setHeader(name, headers[name])
 
+  const requestEtag = req.headers['if-none-match']
+  const bodyEtag = !(body instanceof Stream) ? etag(body || '') : undefined
```

Comment:
The `etag` module doesn't support `Stream`s, and we weren't returning the etag header for `Stream`s/empty body responses anyway, so this seemed like the most straightforward way to work around this.

I'm certainly open to other ideas though.